### ISSUE TITLE
Error: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found

### ISSUE THREAD

**Issue by 0pt1m1z3r** [SalvatorePreviti/roaring-node#54]
The 2.x.x version cause an error. Nodejs 14.19.0.
```
/var/lib/nave/global/14.19.0/lib/node_modules/roaring/index.js:38
    throw e;
    ^

Error: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /var/lib/nave/global/14.19.0/lib/node_modules/roaring/native/roaring-node-v83-linux-x64/roaring.node)
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:1144:18)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:12)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:101:18)
    at /var/lib/nave/global/14.19.0/lib/node_modules/roaring/index.js:31:12
    at Object.<anonymous> (/var/lib/nave/global/14.19.0/lib/node_modules/roaring/index.js:40:3)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32) {
  code: 'ERR_DLOPEN_FAILED'
}
```

```
$ ls -la /lib/x86_64-linux-gnu/libc.so.6
lrwxrwxrwx 1 root root 12 Feb 24  2022 /lib/x86_64-linux-gnu/libc.so.6 -> libc-2.31.so
```

**Comment by SalvatorePreviti** [comment]
I might be wrong but I am afraid this problem is not directly related to roaring-node - I compile and run tests in CI for Ubuntu-latest and it works in CI. For reference, https://github.com/SalvatorePreviti/roaring-node/actions/runs/5002860647/jobs/8963363234

Are you sure you have a recent version of Ubuntu and that your system is properly configured?
Also, whilst I still support older version of node, node 14 is a pretty old version, and I would suggest to think about switching to Node 18 or Node 20.

Could you let me know which version of Ububtu are you using?

**Comment by SalvatorePreviti** [comment]
Yes I am pretty sure is a pretty old version of Ubuntu, that has glibc 3.1, whilst the new roaring-node is compiled against Ubuntu-latest

**Comment by 0pt1m1z3r** [comment]
Ubuntu 20.04.4 LTS. Not so old I think.
Nodejs upgrade is in my plan but for later.
Something was changed between 1.x.x and 2.x.x versions.

**Comment by SalvatorePreviti** [comment]
You have few options.

- update Ubuntu to a newer version
- install the required glibc version
- remove node_modules and run 'npm install --build-from-source' to force roaring-node to be recompiled from source in your machine instead of downloading the precompiled binary

**Comment by SalvatorePreviti** [comment]
Yes a lot of things changed between version 1 and 2, including switching from prebuild to node-pre-gyp - the prebuilt version released is compiled against Ubuntu-latest and I believe this is the reason - it is compiled against a newest version of glibc

**Comment by 0pt1m1z3r** [comment]
I use your module in a project I work with in a big IT company. Unfortunately I can't do much with our environment. Maybe the third option would help me, I will check it.
But anyway would be nice to keep some compatibility for some not much old environment :)
Thank you.

**Comment by SalvatorePreviti** [comment]
I released version 2.2.0 that compiles the prebuilt binaries with Ubuntu 20.04 in CI instead of Ubuntu-latest - this should allow better compatibility with older system.
Can you try if it works for you?

**Comment by 0pt1m1z3r** [comment]
It works! Thank you!
